shared/api/url.go

```diff
@@ -65,3 +66,25 @@ func (u *URL) Target(clusterMemberName string) *URL {
 
 	return u
 }
+
+// UrlsToResourceNames returns a list of resource names extracted from one or more URLs of the same resource type.
+// The resource type path prefix to match is provided by the matchPathPrefix argument.
+func UrlsToResourceNames(matchPathPrefix string, urls ...string) ([]string, error) {
```

Comment:
now we are exporting this function, it should be `URLsToResourceNames` so that acronyms are capitalised.

https://talks.golang.org/2014/names.slide#5

Reply:
Done. PR should be ready to merge now.